### PR TITLE
[new release] ppx_deriving_cad (0.1.0)

### PR DESCRIPTION
**Pull request by geoffder**
PPX Deriver for OCADml transformation functions

- Project page: <a href="https://github.com/OCADml/ppx_deriving_cad">https://github.com/OCADml/ppx_deriving_cad</a>
- Documentation: <a href="https://ocadml.github.io/ppx_deriving_cad/">https://ocadml.github.io/ppx_deriving_cad/</a>

##### CHANGES:

- Initial opam release of `ppx_deriving_cad`

### CODE REVIEW

**Comment by mseri** [comment]
It looks like you are only using OCADml and OSCADml but then those are actually needed at runtime. I think you should do what other derivers, e.g. ppx_deriving_yojson, are doing and add a declaration for your ppx runtime dependencies:
```
 (ppx_runtime_libraries OCADml OSCADml)
```
and require both of them in your opam file (not just as test dependencies.

**Comment by geoffder** [comment]
I didn't think they were required, since I don't actually run anything from either of the libraries during the execution of the derivers. I use them both in the tests to see that generated functions/signatures are actually compatible with the two libraries, but someone should be able to use the ppx with OCADml, without using requiring OSCADml for pre-processing/compilation right? Or am I mistaken about how these thing work?

**Comment by mseri** [comment]
@kit-ty-kate do you know who we could ask?

**Comment by geoffder** [comment]
Also my bad on the concurrent PRs, I didn't realize this one wasn't actually done. That's what I get for releasing three packages at about the same time. I can kill the other tag and just refresh this one when it's figured out.

**Comment by mseri** [comment]
Please don’t do anything for now. It may be that the suggested ch

**Comment by mseri** [comment]
Please don’t do anything for now. It may be that the suggested change is not needed

**Comment by geoffder** [comment]
Sure, I'll have to do the version constraint fix that was in the other release PR before this one gets merged though.

**Comment by geoffder** [comment]
I've finally gotten around to updating this PR and closing the other. Is there any word with respect to the runtime concerns?

**Comment by mseri** [comment]
I am going to merge as is. If with usage we figure out that something is needed i will open an issue in the repo

File: packages/ppx_deriving_cad/ppx_deriving_cad.0.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+maintainer: ["Geoff deRosenroll<geoffderosenroll@gmail.com>"]
+authors: ["Geoff deRosenroll<geoffderosenroll@gmail.com>"]
+license: "GPL-2.0-or-later"
+homepage: "https://github.com/OCADml/ppx_deriving_cad"
+doc: "https://ocadml.github.io/ppx_deriving_cad/"
+bug-reports: "https://github.com/OCADml/ppx_deriving_cad/issues"
+depends: [
+  "dune" {>= "3.2"}
+  "ocaml" {>= "4.14.0"}
+  "base" {>= "0.14.1" & with-test}
```
Comment (mseri) [inline packages/ppx_deriving_cad/ppx_deriving_cad.0.1.0/opam:16]:
```suggestion
  "base" {>= "v0.14.1" & with-test}
```

**Comment by geoffder** [comment]
Ok, sounds good!

**Pull request merged**